src/pyhf/__init__.py

```diff
@@ -27,24 +28,63 @@ def get_backend():
 
 
 @events.register('change_backend')
-def set_backend(backend, custom_optimizer=None):
+def set_backend(backend, custom_optimizer=None, _session=None):
     """
     Set the backend and the associated optimizer
 
     Example:
         >>> import pyhf
         >>> import tensorflow as tf
-        >>> pyhf.set_backend(pyhf.tensor.tensorflow_backend(session=tf.Session()))
+        >>> sess = tf.compat.v1.Session()
+        >>> pyhf.set_backend("tensorflow", _session=sess)
+        >>> pyhf.tensorlib.name
+        'tensorflow'
+        >>> pyhf.set_backend(b"pytorch")
+        >>> pyhf.tensorlib.name
+        'pytorch'
+        >>> pyhf.set_backend(pyhf.tensor.numpy_backend())
+        >>> pyhf.tensorlib.name
+        'numpy'
 
     Args:
-        backend: One of the supported pyhf backends: NumPy, TensorFlow, and PyTorch
+        backend (`str` or `pyhf.tensor` backend): One of the supported pyhf backends: NumPy, TensorFlow, and PyTorch
+        custom_optimizer (`pyhf.optimize` optimizer): Optional custom optimizer defined by the user
+        _session (|tf.compat.v1.Session|_): TensorFlow v1 compatible Session to use when the :code:`"tensorflow"` backend API is used
+
+    .. |tf.compat.v1.Session| replace:: ``tf.compat.v1.Session``
+    .. _tf.compat.v1.Session: https://www.tensorflow.org/api_docs/python/tf/compat/v1/Session
 
     Returns:
         None
     """
     global tensorlib
     global optimizer
 
+    if isinstance(backend, (str, bytes)):
+        if isinstance(backend, bytes):
+            backend = backend.decode("utf-8")
```

Comment:
The backend strings should get put in lowercase.

```suggestion
            backend = backend.decode("utf-8").lower()
```